onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -362,23 +349,16 @@ void BasicBackend::StartAsyncInference(Ort::KernelContext& context, OVInferReque
                                                    infer_request,
                                                    output_name,
                                                    subgraph_context_.output_names);
-        auto allocator_name = tensor.GetTensorMemoryInfo().GetAllocatorName();
-
-        ov_tensor_data_t ov_tensor_data;
-        ort_tensor_key_t ort_tensor_key{tensor.GetTensorRawData(), allocator_name};
-        if (const auto& it = ort_ov_tensor_map.find(ort_tensor_key); it != ort_ov_tensor_map.end()) {
-          ov_tensor_data = it->second;
-        } else {
+        ort_tensor_key_t ort_tensor_key{output_name};
+        const auto& it = ort_ov_tensor_map.find(ort_tensor_key);
+        if ((it == ort_ov_tensor_map.end()) ||
+            (it != ort_ov_tensor_map.end() && (it->second.ort_ptr != tensor.GetTensorRawData()))) {
+          ov_tensor_data_t ov_tensor_data;
           auto output = graph_output_info.at(output_idx);
-          if (allocator_name == OpenVINO_RT_NPU) {
-            ov_tensor_data.copy_needed = false;
-            ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape(),
-                                                                     (void*)tensor.GetTensorRawData());
-          } else {
-            ov_tensor_data.copy_needed = true;
-            ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape());
-          }
-          ort_ov_tensor_map.emplace(ort_tensor_key, ov_tensor_data);
+          ov_tensor_data.ort_ptr = tensor.GetTensorRawData();
+          ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape(),
+                                                                   const_cast<void*>(tensor.GetTensorRawData()));
```

Comment:
non-blocking nit:
Can avoid const_cast by calling `tensor.GetTensorMutableRawData()` on this output tensor. Ort::UnownedValue can return non-const `void*` data: https://github.com/microsoft/onnxruntime/blob/a47254eaef87a4572078d3165f4b9d1c4b4c52b1/include/onnxruntime/core/session/onnxruntime_cxx_api.h#L1484-L1488

Reply:
let's address this in follow up PR

Reply:
Thanks George. I was battling with Git and could not merge the change to address Adrian's comment quickly enough. I'll follow-up with Saurabh to create another PR.